yarn-project/merkle-tree/src/standard_indexed_tree/standard_indexed_tree.ts

```diff
@@ -60,7 +60,7 @@ export const noopDeserializer: FromBuffer<Buffer> = {
  * Standard implementation of an indexed tree.
  */
 export class StandardIndexedTree extends TreeBase<Buffer> implements IndexedTree {
-  #snapshotBuilder = new IndexedTreeSnapshotBuilder(this.store, this, this.leafPreimageFactory);
+  #snapshotBuilder: IndexedTreeSnapshotBuilder;
```

Comment:
wait why?

Reply:
After changing the target, it complains that you're using `this.leafPreimageFactory` before it is initialized

Reply:
I checked the output of `swc` and it reorders it:

```js
export class StandardIndexedTree extends TreeBase {
    leafPreimageFactory;
    leafFactory;
    #snapshotBuilder;
    cachedLeafPreimages;
    leaves;
    leafIndex;
    constructor(store, hasher, name, depth, size = 0n, leafPreimageFactory, leafFactory, root){
        super(store, hasher, name, depth, size, noopDeserializer, root), this.leafPreimageFactory = leafPreimageFactory, this.leafFactory = leafFactory, this.#snapshotBuilder = new IndexedTreeSnapshotBuilder(this.store, this, this.leafPreimageFactory), this.cachedLeafPreimages = {};
        this.leaves = store.openMap(`tree_${name}_leaves`);
        this.leafIndex = store.openMap(`tree_${name}_leaf_index`);
    }
    // ...
```